src/components/FormFields/LabeledOptionsField.js

```diff
@@ -2,20 +2,35 @@ import React from 'react'
 import { PropTypes } from '@dhis2/prop-types'
 import { MultiSelectFieldFF, ReactFinalForm, MultiSelectField } from '@dhis2/ui'
 import i18n from '@dhis2/d2-i18n'
-import { useGetLabeledOptions } from '../../hooks/parameter-options'
+import { useDataQuery } from '@dhis2/app-runtime'
 
 const { Field } = ReactFinalForm
 
+const query = {
+    options: {
+        resource: '/',
+        id: /* istanbul ignore next */ ({ id }) => id,
+        params: {
+            paging: false,
+        },
+    },
+}
+
 /**
  * A labeled options field has options that have both a label and a value,
  * as opposed to the unlabeled options field, where the options just have
  * values.
  */
 
 const LabeledOptionsField = ({ endpoint, label, name, parameterName }) => {
-    const { loading, error, data } = useGetLabeledOptions({
-        endpoint,
-        parameterName,
+    /**
+     * HACK: this is a bit of a hack to allow using the useDataQuery hook with
+     * a dynamic query. Initially we used a custom hook for this but that
+     * replicated all of the internal logic of the useDataQuery hook so this
+     * seems like a better trade-off.
+     */
```

Comment:
I 100% understand why we ultimately ended up with this solution, but it's not optimal, is it?
I feel like we meet this issue in every app-platform app. The `app-runtime` must provide a better solution, otherwise the whole benefit of static declarative queries is gone. Or am I wrong?

Reply:
I agree. I also think that it would be best to address this in the app-runtime. I think a dynamic query is common enough to warrant a solution. Maybe I'm not aware of an existing solution. At any rate, this is definitely not ideal.